packages/nextjs/src/server/index.ts

```diff
@@ -190,8 +191,15 @@ export function init(options: NodeOptions): NodeClient | undefined {
       span.setAttribute(SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN, 'auto');
     }
 
+    // Add headers to the root span if the client options allow it
+    const shouldSendDefaultPii = getClient()?.getOptions().sendDefaultPii ?? false;
+    if (shouldSendDefaultPii && isRootSpan) {
+      const headers = getIsolationScope().getScopeData().sdkProcessingMetadata?.normalizedRequest?.headers;
+      addHeadersAsAttributes(headers, rootSpan);
```

Comment:
The `addHeadersAsAttributes` also checks for `sendDefaultPii`. We could reduce the code (and also the double-checking) here.

You can also change the `addheadersAsAttributes` function to early-return if `sendDefaultPii` is false